tasks/main.yml

```
@@ -33,9 +33,8 @@
       name: awslogs
       state: restarted
       enabled: yes
-  # If amazon linux 1 or not amazon linux 2
-  when: (ansible_distribution != "Amazon" and ansible_os_family != "RedHat" and ansible_distribution_version != "(Karoo)") or
-        (ansible_distribution == "Amazon" and ansible_os_family == "RedHat" and ansible_distribution_version == "2018.03")
+  # If not Amazon Linux 2
+  when: (ansible_distribution != "Amazon" and ansible_os_family != "RedHat" and ansible_distribution_version != "(Karoo)")
```

This isn't quite right. This will not match _any_ Amazon image or _any_ Redhat image. I believe what you want is this:

```suggestion
  when: not (ansible_distribution == "Amazon" and ansible_os_family == "RedHat" and ansible_distribution_version == "(Karoo)")
```

---

I agree with @brainsik although you can also remove the RedHat ansible_os_family check. It's redundant if we're also checking for the Amazon distribution.

---

Now I'm wondering how this ever worked (I guess the OR was doing a lot of work there).